src/webgpu/shader/execution/flow_control/switch.spec.ts

```diff
@@ -0,0 +1,139 @@
+export const description = `
+Flow control tests for switch statements.
+`;
+
+import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import { GPUTest } from '../../../gpu_test.js';
+
+import { runFlowControlTest } from './harness.js';
+
+export const g = makeTestGroup(GPUTest);
+
+g.test('switch')
+  .desc('Test that flow control executes the correct switch case block')
+  .fn(t => {
+    runFlowControlTest(
+      t,
+      f => `
+  ${f.expect_order(0)}
+  switch (${f.value(1)}) {
+    case 0: {
+      ${f.expect_not_reached()}
+      break;
+    }
+    case 1: {
+      ${f.expect_order(1)}
+      break;
+    }
+    case 2: {
+      ${f.expect_not_reached()}
+      break;
+    }
+    default: {
+      ${f.expect_not_reached()}
+      break;
+    }
+  }
+  ${f.expect_order(2)}
+`
+    );
+  });
+
+g.test('switch_multiple_case')
+  .desc(
+    'Test that flow control executes the correct switch case block with multiple cases per block'
+  )
+  .fn(t => {
+    runFlowControlTest(
+      t,
+      f => `
+  ${f.expect_order(0)}
+  switch (${f.value(2)}) {
+    case 0, 1: {
+      ${f.expect_not_reached()}
+      break;
+    }
+    case 2, 3: {
+      ${f.expect_order(1)}
+      break;
+    }
+    default: {
+      ${f.expect_not_reached()}
+      break;
+    }
+  }
+  ${f.expect_order(2)}
+`
+    );
+  });
+
+g.test('switch_multiple_case_default')
+  .desc(
+    'Test that flow control executes the correct switch case block with multiple cases per block (combined with default)'
+  )
+  .fn(t => {
+    runFlowControlTest(
+      t,
+      f => `
+  ${f.expect_order(0)}
+  switch (${f.value(2)}) {
```

Comment:
nit: make this 4, so it's the `default` case which is triggered with the multiple selector

Reply:
That's a bit opinionated. Let's land as is, and add more weird and wonderful edge cases.